src/store/assets/tokens.ts

```diff
@@ -72,11 +73,11 @@ export const nonFungibleTokensState = atom(get => {
 });
 export const stxTokenState = atom(get => {
   const balances = get(accountBalancesState);
-  if (!balances || balances.stx.balance === '0') return;
+  if (!balances || balances.stx.balance.isEqualTo(0)) return;
   return {
     type: 'stx',
     contractAddress: '',
-    balance: balances.stx.balance,
+    balance: new BigNumber(balances.stx.balance),
```

Comment:
much of these conversions can happen where the data is fetched, and it will be passed along automatically.

Reply:
I've come across this problem of "where along the flow of data should the initialisation happen?" before. 

I reckon it's best to keep the data fetching calls as dumb as possible, and do it either in the store (selectors/atoms) or create another layer for this purpose.

This was actually a [question I asked on StackOverflow 4 years ago](https://stackoverflow.com/questions/43520435/where-to-initialise-model-objects-within-a-redux-ngrx-application-architecture) (poorly explained now I read it back). It's in context of redux, but it's still relevant in terms of where along the app lifecycle to do this kind of intermediary data parsing.

I think for now it's alright here, though. Maybe we can consider this more when improving how we handle/pass around currencies (symbol & amount), rather than just amount values.

Reply:
yeah, I agree -- the only thing I was hoping for with the request is that there would be a single step to convert the values from the raw fetch response to bignumber, rather than having to do it in many atoms/components/hooks etc. So maybe we can create a new atom for this purpose?